tools/bin/commands/util/openshift_funcs

```diff
@@ -222,3 +222,42 @@ guess_route() {
+
+    # clean cookie
+    rm -rf "$COOKIE1"
+    touch "$COOKIE1"
+
+    local SYNDESIS_URL="$(oc get route syndesis  --template={{.spec.host}})"
+    # first invocation is needed just to get the redierect url and to define a csrf token
+    local REDIRECTED_URL="$(curl --write-out "%{url_effective}\n" --location --silent --show-error --output /dev/null --insecure --cookie-jar  "$COOKIE1" -v "https://$SYNDESIS_URL" 2>/dev/null )"
+
+    # isolate required params
+    local LOGIN_URL="$(echo $REDIRECTED_URL | sed -e 's/?/\n/g' | head -n1)"
+    local THEN_URL="$(echo $REDIRECTED_URL | sed -e 's/=/\n/g' | tail -n1)"
+    local CSRF=$(cat "$COOKIE1"  | grep -Po '(?<=\scsrf).*' | awk '{$1=$1;print}')
+
+    # login on OpenShift and save output to TEMP_STATE location
+    local TEMP_STATE="$(mktemp /tmp/syndesis_state.XXX.html)"
+    curl --referer "$REDIRECTED_URL" --insecure --location  -H "Cookie: csrf=$CSRF" --data-urlencode username="$USERNAME"  --data-urlencode password="developer" --data-ascii then="$THEN_URL" --data-urlencode csrf="$CSRF" --cookie-jar "$COOKIE2" -i -v -o "$TEMP_STATE" "https://$(minishift ip):8443/login"  2>/dev/null
+
+    # move the freshly created "ssn" cookie entry back to the main cookie
+    tail -1 "$COOKIE2" >> "$COOKIE1"
+
+    # extract state token
+    local STATE=$(grep  -E '"then' $TEMP_STATE | grep -o -E 'state=.*%3' | sed 's#state=##' | sed 's#%3##')
+
+    # approve permission delegation, weaving all the params in the GET parameters
+    curl --insecure --location --cookie-jar "$COOKIE1" --cookie "$COOKIE1" -i -v --data "then=..%2Fauthorize%3Fapproval_prompt%3Dforce%26client_id%3Dsystem%253Aserviceaccount%253Amyproject%253Asyndesis-oauth-client%26redirect_uri%3Dhttps%253A%252F%252F$SYNDESIS_URL%252Foauth%252Fcallback%26response_type%3Dcode%26scope%3Duser%253Ainfo%2Buser%253Acheck-access%26state%3D$STATE%253A%252F&csrf=$CSRF&client_id=system%3Aserviceaccount%3Amyproject%3Asyndesis-oauth-client&user_name=$USERNAME&redirect_uri=https%3A%2F%2F$SYNDESIS_URL%2Foauth%2Fcallback&scope=user%3Ainfo&scope=user%3Acheck-access&approve=Allow+selected+permissions"  -H 'dnt: 1' -H "referer: https://$(minishift ip):8443/oauth/authorize/approve?client_id=system%3Aserviceaccount%3Amyproject%3Asyndesis-oauth-client&redirect_uri=https%3A%2F%2F$SYNDESIS_URL%2Foauth%2Fcallback&scope=user%3Ainfo+user%3Acheck-access&then=..%2Fauthorize%3Fapproval_prompt%3Dforce%26client_id%3Dsystem%253Aserviceaccount%253Amyproject%253Asyndesis-oauth-client%26redirect_uri%3Dhttps%253A%252F%252F$SYNDESIS_URL%252Foauth%252Fcallback%26response_type%3Dcode%26scope%3Duser%253Ainfo%2Buser%253Acheck-access%26state%3D$STATE%253A%252F" "https://$(minishift ip):8443/oauth/authorize/approve" -o /dev/null 2>/dev/null
+
+    # smoke test showing how to use the cookie
+    echo "Cookie created at $COOKIE1"
+    echo Sample invocation: curl -k --cookie "$COOKIE1" "https://$(oc get route syndesis --template={{.spec.host}})/api/v1/connections"
```

Comment:
That would need to go away if we are going to print out the cookie to stdout like suggested above.